cmd/limactl/guest_install.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/lima-vm/lima/pkg/cacheutil"
+	"github.com/lima-vm/lima/pkg/store"
+	"github.com/lima-vm/lima/pkg/store/filenames"
+	"github.com/lima-vm/lima/pkg/usrlocalsharelima"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func newGuestInstallCommand() *cobra.Command {
+	guestInstallCommand := &cobra.Command{
+		Use:               "guest-install INSTANCE",
+		Short:             "Install guest components",
+		Args:              WrapArgsError(cobra.MaximumNArgs(1)),
+		RunE:              guestInstallAction,
+		ValidArgsFunction: cobra.NoFileCompletions,
+		Hidden:            true,
+	}
+	guestInstallCommand.Flags().BoolP("rsync", "", false, "use rsync")
+	return guestInstallCommand
+}
+
+func runCmd(name string, flags []string, args ...string) error {
+	cmd := exec.Command(name, append(flags, args...)...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	logrus.Debugf("executing %v", cmd.Args)
+	return cmd.Run()
+}
+
+func guestInstallAction(cmd *cobra.Command, args []string) error {
+	rsync, err := cmd.Flags().GetBool("rsync")
+	if err != nil {
+		return err
+	}
+
+	instName := DefaultInstanceName
+	if len(args) > 0 {
+		instName = args[0]
+	}
+
+	inst, err := store.Inspect(instName)
+	if err != nil {
+		return err
+	}
+	if inst.Status == store.StatusStopped {
+		return fmt.Errorf("instance %q is stopped, run `limactl start %s` to start the instance", instName, instName)
+	}
+
+	sshExe := "ssh"
+	sshConfig := filepath.Join(inst.Dir, filenames.SSHConfig)
+	sshFlags := []string{"-F", sshConfig}
+
+	scpExe := "scp"
+	scpFlags := sshFlags
+	if rsync {
+		scpExe = "rsync"
+		scpFlags = []string{"-e", "ssh -F " + sshConfig, "--progress"}
+	}
+
+	y, err := inst.LoadYAML()
+	if err != nil {
+		return err
+	}
+
+	hostname := fmt.Sprintf("lima-%s", inst.Name)
+	prefix := *y.GuestInstallPrefix
+
+	// lima-guestagent
+	guestAgentBinary, err := usrlocalsharelima.GuestAgentBinary(*y.OS, *y.Arch)
+	if err != nil {
+		return err
+	}
+	tmp := "/tmp/lima-guestagent"
```

Comment:
Generically this should use a random path, or a path under `/root`. Otherwise a non-root user on the guest can hijack the path with a race condition. But practically it is not a problem for Lima, at least until Lima implements a remote SSH driver.

Reply:
Using temporary files means that the rsync hack will not work, so needs another mechanism for delta updates. Probably something that is more similar to the version checking, that the cidata scripts are currently doing.